tests/test_regression.py

```diff
@@ -0,0 +1,47 @@
+import unittest
+from skfda.representation.basis import Monomial, Fourier, FDataBasis
+from skfda.ml.regression import LinearScalarRegression
+import numpy as np
+
+class TestRegression(unittest.TestCase):
+    """Test regression"""
+
+    def test_linear_scalar_regression_auto(self):
+        beta_basis = Fourier(nbasis=5)
+        beta_fd = FDataBasis(beta_basis, [1, 2, 3, 4, 5])
+
+        x_basis = Monomial(nbasis=7)
+        x_fd = FDataBasis(x_basis, np.identity(7))
+
+        scalar_test = LinearScalarRegression([beta_fd])
+        y = scalar_test.predict([x_fd])
```

Comment:
Calling `fit` is always required before `predict`. You can call both with the same data using `fit_predict`.